kmodes/kprototypes.py

```diff
@@ -364,24 +376,25 @@ def _k_prototypes_single(Xnum, Xcat, nnumattrs, ncatattrs, n_clusters, n_points,
         # can do k-means on the numerical attributes.
         cl_attr_sum = np.zeros((n_clusters, nnumattrs), dtype=np.float64)
         # Same for the membership sum per cluster
-        cl_memb_sum = np.zeros(n_clusters, dtype=int)
+        cl_memb_sum = np.zeros(n_clusters, dtype=np.float64)
         # cl_attr_freq is a list of lists with dictionaries that contain
         # the frequencies of values per cluster and attribute.
-        cl_attr_freq = [[defaultdict(int) for _ in range(ncatattrs)]
+        cl_attr_freq = [[defaultdict(float) for _ in range(ncatattrs)]
                         for _ in range(n_clusters)]
         for ipoint in range(n_points):
+            sample_weight = sample_weights[ipoint] if sample_weights is not None else 1
             # Initial assignment to clusters
             clust = np.argmin(
                 num_dissim(centroids[0], Xnum[ipoint]) + gamma *
                 cat_dissim(centroids[1], Xcat[ipoint], X=Xcat, membship=membship)
             )
             membship[clust, ipoint] = 1
-            cl_memb_sum[clust] += 1
+            cl_memb_sum[clust] += sample_weight
```

Comment:
Ultimately, we calculate the mean by dividing the `cl_attr_sum` by this, the `cl_mem_sum`: https://github.com/nicodv/kmodes/blob/master/kmodes/kprototypes.py#L471

If we apply the weight to both the numerator and denominator, they cancel out, no?

Shouldn't we solely apply the weight to `cl_attr_sum`?

Reply:
I do wonder now how the unit test that tests for a single overweighted sample was able to pass. 🤔

Reply:
Yeah, so do I. I think the problem was only for one of numerical/categorical features. Maybe the other having been correct was sufficient to push the centroid to the right point?